### ISSUE TITLE
The use of `useLayoutEffect` causes a warning

### ISSUE THREAD

**Issue by pedropalau** [richardscarrott/react-snap-carousel#12]
This is the resulting warning message from using `useLayoutEffect` when running in the server:

```
Warning: useLayoutEffect does nothing on the server, because its effect cannot be encoded into the server renderer's output format. This will lead to a mismatch between the initial, non-hydrated UI and the intended UI. To avoid this, useLayoutEffect should only be used in components that render exclusively on the client. See https://reactjs.org/link/uselayouteffect-ssr for common fixes.
```

This can be easily fixed with something like:

```
const useIsomorphicLayoutEffect = typeof window !== 'undefined' ? useLayoutEffect : useEffect
```

I would be more than glad to open a PR.

**Comment by loureirorg** [comment]
I'm having the same issue. 

An easy way to reproduce this issue is to create a NextJS project and add the "react-snap-carousel" library.

NextJS has SSR by default, which triggers the issue.

**Comment by richardscarrott** [comment]
Fixed by @pedropalau https://github.com/richardscarrott/react-snap-carousel/pull/13 🙏

Published latest tag as 0.3.2 `npm add react-snap-carousel@0.3.2`. I tested in Remix but please let me know if you run into any issues 👍
